src/main/java/edu/ksu/canvas/attendance/services/CanvasApiWrapperService.java

```diff
@@ -140,7 +140,11 @@ void setEnrollmentOptionsFactory(EnrollmentOptionsFactory enrollmentOptionsFacto
     public Optional<Assignment> getSingleAssignment(Long courseId, OauthToken oauthToken, String assignmentId) throws IOException {
         AssignmentReader assignmentReader = canvasApiFactory.getReader(AssignmentReader.class, oauthToken);
         GetSingleAssignmentOptions singleAssignmentOptions = new GetSingleAssignmentOptions(courseId.toString(), assignmentId);
-        return assignmentReader.getSingleAssignment(singleAssignmentOptions);
+        try{
+            return assignmentReader.getSingleAssignment(singleAssignmentOptions);
+        } catch(Exception e){
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1166)